### PR TITLE
[Docs] Update the tutorial links

### PR DESCRIPTION
**Pull request by yhyakuna**
### :hammer_and_wrench: Description

**Docs change only**

This PR:

- Add additional tutorial links where available
- Replace all tutorials links to point to the new HashiCorp Developer site

### :ship: Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-hcp/blob/main/CHANGELOG.md):

```release-note
NONE
```

### :building_construction: Acceptance tests

- [ ] Are there any feature flags that are required to use this functionality?
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: docs/resources/vault_cluster.md [hunk 1]
```diff
@@ -49,11 +49,11 @@ resource "hcp_vault_cluster" "example" {
...
-- `metrics_config` (Block List, Max: 1) The metrics configuration for export. (https://learn.hashicorp.com/tutorials/cloud/vault-metrics-guide#metrics-streaming-configuration) (see [below for nested schema](#nestedblock--metrics_config))
+- `metrics_config` (Block List, Max: 1) The metrics configuration for export. (https://developer.hashicorp.com/vault/tutorials/cloud-monitoring/vault-metrics-guide#metrics-streaming-configuration) (see [below for nested schema](#nestedblock--metrics_config))
 - `min_vault_version` (String) The minimum Vault version to use when creating the cluster. If not specified, it is defaulted to the version that is currently recommended by HCP.
-- `paths_filter` (List of String) The performance replication [paths filter](https://learn.hashicorp.com/tutorials/vault/paths-filter). Applies to performance replication secondaries only and operates in "deny" mode only.
+- `paths_filter` (List of String) The performance replication [paths filter]https://developer.hashicorp.com/vault/tutorials/cloud-ops/vault-replication-terraform). Applies to performance replication secondaries only and operates in "deny" mode only.
```
Comment (i0rek) [inline docs/resources/vault_cluster.md:56]:
```suggestion
- `paths_filter` (List of String) The performance replication [paths filter](https://developer.hashicorp.com/vault/tutorials/cloud-ops/vault-replication-terraform#review-hcpvault-tf). Applies to performance replication secondaries only and operates in "deny" mode only.
```

there is a `(` missing and above you linked to a specific section with a `paths_filter` example. I assume this is what should be in here as well. I might be wrong though.

**Review by i0rek — APPROVED** [review]
LGTM, thanks for doing this. Found a small issue.

**Review by bcmdarroch — COMMENTED** [review]
Changes all look good- but they need to made in the template/resource files and be regenerated 😅 I know our template directory well so I can apply these changes

File: docs/index.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ description: |-
 
 -> Please refer to the provider's [Release Notes](https://github.com/hashicorp/terraform-provider-hcp/releases) for critical fixes.
 
--> Consul on Azure is now available in public beta. [Get started with end-to-end deployment configuration](https://learn.hashicorp.com/tutorials/cloud/consul-end-to-end-overview).
+Consul on Azure is available. See the [Get started with end-to-end deployment configuration](https://developer.hashicorp.com/consul/tutorials/cloud-deploy-automation/consul-end-to-end-overview) tutorial.
```
Comment (bcmdarroch) [inline docs/index.md:12]: This banner is kind of out-of-date anyway, so could drop this entirely.

**Review by bcmdarroch — APPROVED** [review]

**Pull request merged**